### PR TITLE
Private Cloud Operator release February 18

### PR DESCRIPTION
**Pull request by zlogic**
Updated documentation, release notes and upgrade instructions for the new Mendix Operator version.

We're planning to release this new version on Thursday, ~~February 11~~ February 18.

### CODE REVIEW

**Review by MarkvanMents — COMMENTED** [review]
Hi Dmitrii (@zlogic )
I've proofread this, but I have a number of questions - particularly about the user and management accounts which seem to be new concepts which aren't fully explained and could be confusing.

Can you respond to my comments and let me know if this is OK.

**Comment by zlogic** [comment]
Hi @MarkvanMents, thank you for your comments! I'll go through the document again and rephrase/fix any issues.

We've postponed the release for one week, so the planned release date is now February 18.

**Comment by MarkvanMents** [comment]
Thanks for the update.

File: content/releasenotes/developer-portal/mendix-for-private-cloud.md [hunk 1]
```diff
@@ -13,6 +13,20 @@ For information on the current status of deployment to Mendix for Private Cloud
...
 ## 2021
 
+### February 11th, 2021
+
+#### Mendix for Private Cloud — Mendix Operator v1.8.0 and Mendix Gateway Agent v1.7.0
+
+* We have improved dependency version management, so upgrading to future versions of the Mendix Operator will be easier.
+* We have introduced versioning for all components, instead of using the "latest" version. All components are now tested for compatibility, and once the Mendix Operator is installed, its behavior will remain unchanged. Components can only be updated through a manual update process.
+* We have added a feature to report installed component versions to the Private Cloud Portal. In the future, this will be used to indicate which Mendix for Private Cloud Operator version is installed and which features it supports.
+* We have added a new S3 option, allowing the use of an existing IAM policy and S3 bucket. Each environment will still get its own user and will not have access to files from other environments.
```
[the remark this answers is not in the document]
Comment (zlogic) [reply content/releasenotes/developer-portal/mendix-for-private-cloud.md:23]: I was trying to follow [Amazon's terminology](https://aws.amazon.com/iam/):

* _IAM user_ is an "account" 
* AWS Account is a billing term, most organizations have just one AWS account for the whole company.

Would `will still get its own account (AWS IAM user)` be a better alternative?

[the remark this answers is not in the document]
Comment (MarkvanMents) [reply content/releasenotes/developer-portal/mendix-for-private-cloud.md:23]: Good suggestion - I'll make that change.

File: content/developerportal/deploy/private-cloud-cluster.md [hunk 2]
```diff
@@ -337,7 +341,243 @@ If the MinIO URL is specified with an `http` schema, TLS will not be used.
...
+{{% alert type="info" %}}
+To use this plan, [upgrade](/developerportal/deploy/private-cloud-upgrade-guide) the Mendix Operator to version 1.8.0 or later.
+{{% /alert %}}
+
+**S3 (create bucket and user with existing policy)** will connect to an AWS account to create S3 buckets and associated IAM accounts. Each app will receive a dedicated S3 bucket and an IAM account. The specified existing policy will be attached to the account. You will need to provide all the information about your Amazon S3 storage such as plan name, region, policy ARN, access key, and secret key.
+
+To enable this mode, select the following options: **Create Bucket**, **Create User**.
+
+Create an IAM policy that will be attached to app users and write down its Policy ARN (specify this value in the **Attach Policy ARN** field):
```
[the remark this answers is not in the document]
Comment (zlogic) [reply content/developerportal/deploy/private-cloud-cluster.md]: We've renamed those UI options to **Create S3 bucket per environment** and **Create account (IAM user) per environment** and updated the documentation pull request. Can you please check if it's clearer now?

[the remark this answers is not in the document]
Comment (zlogic) [reply content/developerportal/deploy/private-cloud-cluster.md]: I've also went through the S3 configuration options and replaced **user** with **IAM user account**.

File: content/developerportal/deploy/private-cloud-cluster.md [hunk 3]
```diff
@@ -299,7 +299,11 @@ To use TLS, specify the MinIO URL with an `https` schema, for example `https://m
 If the MinIO URL is specified with an `http` schema, TLS will not be used.
 {{% /alert %}}
 
-**S3 (create on-demand)** will connect to an AWS account to create S3 buckets and associated IAM accounts. Each app will receive a dedicated S3 bucket and an IAM account which only has access to that specific S3 bucket. You will need to provide all the information about your Amazon S3 storage such as plan name, region, access key, and secret key. The associated IAM account needs to have the following IAM policy (replace `<account_id>` with your AWS account number):
+**S3 (create bucket and user with inline policy)** will connect to an AWS account to create S3 buckets and associated IAM accounts. Each app will receive a dedicated S3 bucket and an IAM account with an inline policy only has access to that specific S3 bucket. You will need to provide all the information about your Amazon S3 storage such as plan name, region, access key, and secret key.
+
+To enable this mode, select the following options: **Create Bucket**, **Create User**, **Create inline policy**.
+
+The management IAM account needs to have the following IAM policy (replace `<account_id>` with your AWS account number):
```
[the remark this answers is not in the document]
Comment (zlogic) [reply content/developerportal/deploy/private-cloud-cluster.md]: I've added a note what this account means:

> The Mendix Operator will use a **management IAM user account** to create and delete S3 buckets and IAM user accounts.

> The Mendix Operator will use a **management IAM user account** to create and delete IAM user accounts.

File: content/developerportal/deploy/private-cloud-cluster.md [hunk 4]
```diff
@@ -337,7 +341,243 @@ If the MinIO URL is specified with an `http` schema, TLS will not be used.
...
+
+{{% alert type="info" %}}
+If the plan name already exists you will receive an error that it cannot be created. This is not a problem, you can continue to use the plan, and it will now have the new configuration.
+{{% /alert %}}
+
+{{% alert type="info" %}}
+To use this plan, [upgrade](/developerportal/deploy/private-cloud-upgrade-guide) the Mendix Operator to version 1.8.0 or later.
+{{% /alert %}}
+
+**S3 (create user with inline policy)** will connect to an AWS account to create IAM accounts. Each app will receive a dedicated IAM account with an inline policy which only has access to a subdirectory in the existing S3 bucket. You will need to provide all the information relating to your Amazon S3 storage such as plan name, bucket name, region, access key, and secret key.
```
[the remark this answers is not in the document]
Comment (zlogic) [reply content/developerportal/deploy/private-cloud-cluster.md]: I agree, "folder" is something only visible in the AWS web console, and "prefix" is a much better term that's used everywhere else in the AWS documentation. Can you please check if the updated sentence makes more sense?

File: content/developerportal/deploy/private-cloud-upgrade-guide.md [hunk 5]
```diff
@@ -125,11 +139,10 @@ kubectl -n $OPERATOR_NAMESPACE get storageplan --no-headers=true -o name | sed -
   xargs -I {} kubectl -n $OPERATOR_NAMESPACE patch storageplan {} --type=merge -p '{"spec":{"type":"on-demand"}}'
 ```
 
-##### 2.2.4.2 Updating the Mendix Operator Configuration (from versions 1.1.\*, 1.2.\*, 1.3.\*, and 1.4.\*){#update-configuration-1.1.0}
+##### 2.2.5.2 Updating the Mendix Operator Configuration (from versions 1.1.\*, 1.2.\*, 1.3.\*, and 1.4.\*){#update-configuration-1.1.0}
 
 {{% alert type="info" %}}
-Follow this step only when upgrading from Mendix Operator 1.1.\*, 1.2.\*, 1.3.\*, and 1.4.\*.
-
+Follow this step only when upgrading from Mendix Operator 1.1.\*-1.7.\* only.
```
[the remark this answers is not in the document]
Comment (zlogic) [reply content/developerportal/deploy/private-cloud-upgrade-guide.md]: Thanks for noticing - I've fixed this alert message with a new commit.

**Review by MarkvanMents — APPROVED** [review]
Hi Dmitrii,
@zlogic 
Have just caught one place where I think it should have been an IAM user account and not just "user". But now ready to publish.
OK to go ahead and publish?

**Comment by zlogic** [comment]
Hi @MarkvanMents, good catch! Indeed, that was a very misleading sentence :)

The changes are OK to publish at any time.

**Pull request merged**